### PR TITLE
feat: init event

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
context: https://www.notion.so/walletconnect/Walletkit-events-sdk-init-event-08b51439dd8b42c5bbe993c63d2ce54e

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
tests
## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

File: packages/core/src/controllers/events.ts [hunk 1]
```diff
@@ -39,6 +39,32 @@ export class EventClient extends IEventClient {
     );
   }
 
+  public init: IEventClient["init"] = async () => {
+    if (typeof process !== "undefined" && process.env.IS_VITEST === "true") return;
```
Comment (bkrem) [inline packages/core/src/controllers/events.ts]: Why is this needed? 

Also I believe we have this exact check somewhere else already so if we need it in multiple places it's safer to abstract it into a util fn; avoids e.g. forgetting the `typeof` check and/or updating lots of inline checks if we ever need to change this.

**Review by bkrem — APPROVED** [review]

File: packages/core/src/controllers/events.ts (hunk 1 above)
Comment (ganchoradkov) [reply packages/core/src/controllers/events.ts]: good idea to make it an util.
 >Why is this needed?

so we don't spam logs on dev/ci runs

File: packages/core/src/controllers/events.ts [hunk 2]
```diff
@@ -39,6 +39,32 @@ export class EventClient extends IEventClient {
...
+    try {
+      const initEvent = {
+        eventId: uuidv4(),
+        bundleId: this.core.projectId || "",
```
Comment (Cali93) [inline packages/core/src/controllers/events.ts]: `bundleId` should be used by mobile with app's bundle ID or package Name.
`domain` should be used for web with the current app's domain.

If none is available to you, you can remove both `bundleId` and `domain` and we'll generate one at the API level.

File: packages/core/src/controllers/events.ts [hunk 3]
```diff
@@ -39,6 +39,32 @@ export class EventClient extends IEventClient {
...
+        timestamp: Date.now(),
+        props: {
+          event: "INIT",
+          type: "",
```
Comment (Cali93) [inline packages/core/src/controllers/events.ts:50]: cc @dnul if you're relying on the type here, what should the value be ?

File: packages/core/src/controllers/events.ts [hunk 4]
```diff
@@ -189,4 +209,15 @@ export class EventClient extends IEventClient {
       this.logger.warn(error);
     }
   };
+
+  private sendPost = async (events: EventClientTypes.Event[]) => {
```
Comment (Cali93) [inline packages/core/src/controllers/events.ts]: nit
```suggestion
  private sendEvent = async (events: EventClientTypes.Event[]) => {
```

**Review by Cali93 — APPROVED** [review]
LGTM given that we address the `bundleId` comment. Did you align on the format with the data team?

File: packages/core/src/controllers/events.ts (hunk 3 above)
Comment (ganchoradkov) [reply packages/core/src/controllers/events.ts:50]: as per the specs https://www.notion.so/walletconnect/Walletkit-events-sdk-init-event-08b51439dd8b42c5bbe993c63d2ce54e?pvs=4#49045ac3ba8d4f36900a24047b9cc87b
```
val event: String = "INIT"
val type: String, // None
```
